### ISSUE TITLE
Segmentation fault  when running python coco_test.py with Large Scale Test-bed included

### ISSUE THREAD

**Issue by oaelhara** [numbbo/coco#1103]
When testing the python code that includes the large-scale test-bed, the test:
`python do.py test-python`
fails, more specifically:
`python coco_test.py bbob2009_testcases.txt` 
(or just  `python coco_test.py`)
we get:

> Running doctests...
>   doctest of <module 'cocoex' from '/Users/wassimaitelhara/.local/lib/python2.7/site-packages/cocoex-0.7.beta.3-py2.7-macosx-10.5-x86_64.egg/cocoex/__init__.pyc'>
>   CAVEAT: doctest OF cocoex.interface IS, FOR SOME REASON, INEFFECTIVE IN PYTHON 2 
>   doctest of <module 'cocoex.interface' from '/Users/wassimaitelhara/.local/lib/python2.7/site-packages/cocoex-0.7.beta.3-py2.7-macosx-10.5-x86_64.egg/cocoex/interface.so'>
> COCO INFO: Results will be output to folder exdata/doctest-001
> COCO INFO: Results will be output to folder exdata/random_search_on_bbob2009-001
> COCO INFO: Results will be output to folder exdata/default
> Segmentation fault: 11

the error comes form `testmod(interface)` in `run_doctests()`  of 
`code-experiments/build/python/coco_test.py`

**Comment by NDManh** [comment]
I think this issue comes from checking commands inside the variable `__pyx_k_observe_problem_let_self_observ[] = "........."` (line 1626 in the file `code-experiments/build/python/cython/interface.c`, branch `level-test3`). I try to run all suggested command and obtain result as follows:
```
pc30:numbbo NGUYEN$ python
Python 2.7.11 |Anaconda 2.5.0 (x86_64)| (default, Dec  6 2015, 18:57:58) 
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import cocoex as ex
>>> suite = ex.Suite("bbob", "", "")
>>> assert len(suite) == 2160
>>> f = suite.get_problem(33)
>>> assert isinstance(f, Problem)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'Problem' is not defined
>>> assert f.id.endswith('f003_i04_d02')
>>> observer = ex.Observer("bbob", "").observe(f)
COCO INFO: Results will be output to folder exdata/default-001
>>> f.free()
Segmentation fault: 11
pc30:numbbo NGUYEN$ 
``` 
I got two errors: one for `assert isinstance(f, Problem)`; and one for `f.free()` with the description `Segmentation fault: 11` as in the issue. But I do not know how to fix them. If I remove all suggested command inside the variable `__pyx_k_observe_problem_let_self_observ[]`, for instance:
```
__pyx_k_observe_problem_let_self_observ[] = "";
```
then the command `python do.py test-python` back to normal.

**Comment by NDManh** [comment]
Also, I tried to do something below on the latest version of branch `development` for tracking the error when running `testmod` on module `interface`.
1) I don't have any problem when running the comment `python do.py test-python`. However I get an error with the following commands:
```
client-172-18-5-184:coco NGUYEN$ python
Python 2.7.11 |Anaconda 2.5.0 (x86_64)| (default, Dec  6 2015, 18:57:58) 
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import cocoex as ex
>>> interface = ex.interface
>>> import doctest
>>> doctest.testmod(interface,raise_on_error=True)
COCO INFO: Results will be output to folder exdata/doctest-002
COCO INFO: Results will be output to folder exdata/random_search_on_bbob2009-002
COCO INFO: Results will be output to folder exdata/default-006
Segmentation fault: 11
```
2) I think this error is due to the text ` >>> observer = ex.Observer(\"bbob\", \"\").observe(f)\n` in the  line `static char __pyx_k_observe_problem_let_self_observ[] ="...."` (line 1626 in the file `code-experiments/build/python/cython/interface.c`) because if I remove just this text and remain the others of this line then these commands above will run through without error:
```
>>> import cocoex as ex
>>> interface = ex.interface
>>> import doctest
>>> doctest.testmod(interface,raise_on_error=True)
COCO INFO: Results will be output to folder exdata/doctest-003
COCO INFO: Results will be output to folder exdata/random_search_on_bbob2009-003
TestResults(failed=0, attempted=62)
```
My question is: why don't we see the error when running the command `python do.py test-python` although this command will call the command `testmod(interface)` (line 93 in `code-experiments/build/python/coco_test.py`) inside it?

**Comment by brockho** [comment]
There is another issue reporting a segmentation fault (#687). Maybe this is related?

**Comment by NDManh** [comment]
This issue is because we need to check whether `(logger->index_file != NULL) ` before the command (lines 363-364 in the file `logger_bbob.c`)
```
 fprintf(logger->index_file, ":%ld|%.1e", logger->number_of_evaluations,
            logger->best_fvalue - logger->optimal_fvalue);
```
And I have fixed it in the full request #1173.

**Comment by oaelhara** [comment]
This issue can be closed
